install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -3816,6 +3816,9 @@ clustermesh:
       # -- base64 encoded PEM values for the clustermesh-apiserver server certificate and private key.
       # Used if 'auto' is not enabled.
       server:
+        # -- Name of the Secret containing the clustermesh-apiserver server certificate and private key.
+        # If specified, cert and key are ignored.
+        existingSecret: ""
         cert: ""
         key: ""
```

Comment:
If you are up for it, I'd take the opportunity to deprecate the `cert` and `key` fields, similarly to what Hubble did, so that we can finally get rid of them during the next cycle. They are totally insecure nonetheless. 

The `enableSecrets` option above could be also deprecated, as superseded by the `existingSecret` configurations that you just added.

Reply:
Ohh yes I will definitely do that, I overlooked this and wasn't aware that it was deprecated actually :eyes: